### ISSUE TITLE
A number of calendar chart examples are broken

### ISSUE THREAD

**Issue by jonavila** [ecomfe/echarts#6454]
### One-line summary [问题简述]
On the website demos, a couple of calendar examples are broken:
https://ecomfe.github.io/echarts-examples/public/editor.html?c=calendar-effectscatter
https://ecomfe.github.io/echarts-examples/public/editor.html?c=calendar-vertical
https://ecomfe.github.io/echarts-examples/public/editor.html?c=calendar-horizontal

### Version & Environment [版本及环境]
+ ECharts version [ECharts 版本]:
+ Browser version [浏览器类型和版本]:
+ OS Version [操作系统类型和版本]:

### ECharts option [ECharts配置项]

```javascript
option = {

}

```

**Comment by leogariba** [comment]
I am having some wired problems with calendar also. This problem happens in every browser of my machine (Windows 10 x64). But not on my VM (Linux, Windows 10 and Android).

I got this error when I try to render my chart: Uncaught TypeError: Cannot read property '0' of undefined
    at Object.buildPath .

Trying to debug the problem, I realized that are missing some points in the shape.points array (mising 4 and 5)
![error](https://user-images.githubusercontent.com/3786863/29490011-c670239a-84e3-11e7-9578-49ddbc1f458b.png)

But I cant figure out  why this points are missing in the first place.
